### ISSUE TITLE
[FEA] PyTorch model with customizable output dim

### ISSUE THREAD

**Issue by mzanotto** [NVIDIA-Merlin/NVTabular#1203]
**Is your feature request related to a problem? Please describe.**
I am using NVTabular to train a PyTorch model. In the current implementation (https://github.com/NVIDIA-Merlin/NVTabular/blob/main/nvtabular/framework_utils/torch/models.py) the Model's output dimension is fixed to 1. In some application scenarios it would be beneficial to be able to set it to an arbitrary dimension (e.g. using the output nodes to learn the parameters of a distribution, learn an embedding of the inputs with Deep Metric Learning frameworks, etc.).

**Describe the solution you'd like**
It would be great to be able to pass the output dimension as a parameter.

**Describe alternatives you've considered**
Current class can be sub-classed to achieve that.

**Additional context**
Not relevant.
